modules/heat_conduction/test/tests/view_factors/tests

```diff
@@ -1,22 +1,64 @@
 [Tests]
-  issues = '#13918'
-  [./view_factor_cube]
+  issues = '#13918 #16954'
+
+  [unnormalized]
     type = CSVDiff
-    input = 'view_factor_cube.i'
-    csvdiff = 'view_factor_cube_out.csv'
-    cli_args = 'UserObjects/view_factor/normalize_view_factor=false'
+    input = 'view_factor_3d.i'
+    csvdiff = 'view_factor_3d_unnormalized_out.csv'
+    cli_args = 'Outputs/file_base=view_factor_3d_unnormalized_out
+                UserObjects/unobstructed_vf/normalize_view_factor=false'
     design = 'UnobstructedPlanarViewFactor.md'
     mesh_mode = REPLICATED
-    requirement = 'The system shall compute view factors for unobstructed, planar surfaces in radiative exchange'
-  [../]
+    requirement = 'The system shall compute view factors for unobstructed, planar surfaces without normalization.'
+  []
+
+  [obstructed]
+    type = CSVDiff
+    input = 'view_factor_obstructed.i'
+    csvdiff = 'view_factor_obstructed_out.csv'
+    design = 'RayTracingViewFactor.md'
+    requirement = 'The system shall compute view factors for cavities with obstruction using ray tracing.'
+  []
 
-  [./view_factor_cube_normalized]
+  [analytical2D]
     type = CSVDiff
-    input = 'view_factor_cube.i'
-    csvdiff = 'view_factor_cube_normalized.csv'
-    cli_args = 'Outputs/file_base=view_factor_cube_normalized'
+    input = 'view_factor_2d.i'
+    csvdiff = 'view_factor_2d_out.csv'
     design = 'UnobstructedPlanarViewFactor.md'
     mesh_mode = REPLICATED
-    requirement = 'The system shall compute view factors for unobstructed, planar surfaces in radiative exchange and normalize view factors from each surface to sum to one'
-  [../]
+    requirement = 'The system shall compute view factors for unobstructed, planar surfaces in two-dimensional meshes using simple quadrature rules'
+  []
+
+  [ray2D]
+    type = CSVDiff
+    input = 'view_factor_2d.i'
+    csvdiff = 'view_factor_2d_ray_out.csv'
+    cli_args = 'Outputs/file_base=view_factor_2d_ray_out
+                UserObjects/active=\'vf_study rt_vf\'
+                RayBCs/active=vf
+                GlobalParams/view_factor_object_name=rt_vf'
+    design = 'RayTracingViewFactor.md'
+    requirement = 'The system shall compute view factors for unobstructed, planar surfaces in two-dimensional meshes using ray tracing'
+  []
+
+  [analytical3D]
+    type = CSVDiff
+    input = 'view_factor_3d.i'
+    csvdiff = 'view_factor_3d_out.csv'
+    design = 'UnobstructedPlanarViewFactor.md'
+    mesh_mode = REPLICATED
```

Comment:
Please expand on this comment.

Reply:
just asking why you need the replicated mesh

Reply:
Because unobstructed planar view factor can query an arbitrary remote side for its info and the communication to make this happen just isn't implemented. This object is basically a double spatial integral over an i element side and a j element side. The j element side can generally be non-local though.